### PR TITLE
Column compression polishing

### PR DESCRIPTION
**Pull request by neyrox**
### Changelog entry 

More column compression tests and corner cases fixes

### Changelog category 

* Not for changelog (changelog entry is not required)

### Description for reviewers 

WIP

### CODE REVIEW

**Review by dorooleg — DISMISSED** [review]

File: ydb/tests/olap/column_compression/fast/test_incorrect.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+        cls.test_dir = f"{cls.ydb_client.database}/{cls.test_name}"
+
+    COMPRESSION_CASES = [
+        ("just_text",         'qwerty',                         "mismatched input"),
+        ("unknown_key",       'codec=lz4',                      "Only algorithm and level settings supported"),
+        ("unknown_algorithm", 'algorithm=rar',                  "Unknown compression algorithm"),
+        ("only_level",        'level=3',                        "specified without an algorithm"),
+        ("negative_level",    'algorithm=zstd, level=-1',       "extraneous input"),
+        ("too_big_level",     'algorithm=zstd, level=99',       "annot parse serializer"),
```
[the remark this answers is not in the document]
Comment (neyrox) [reply ydb/tests/olap/column_compression/fast/test_incorrect.py:41]: This is done intetionally, because of different cases of "c" in different tests

File: ydb/core/kqp/provider/yql_kikimr_exec.cpp [hunk 2]
```diff
@@ -1517,6 +1517,11 @@ bool ParseCompressionSettings(
             }
         } else if (key == "level" && (settingVal.IsCallable("Uint64") || settingVal.IsCallable("Int64"))) {
             compression->set_compression_level(FromString<i64>(settingVal.Child(0)->Content()));
+            if (!compression->has_algorithm()) {
+                ctx.AddError(TIssue(ctx.GetPosition(settingVal.Pos()), TStringBuilder()
+                    << "compression level specified without an algorithm"));
+                return false;
+            }
```
[the remark this answers is not in the document]
Comment (neyrox) [reply ydb/core/kqp/provider/yql_kikimr_exec.cpp]: This can be done, but should it?

[the remark this answers is not in the document]
Comment (dorooleg) [reply ydb/core/kqp/provider/yql_kikimr_exec.cpp]: дело говорит

**Comment by dorooleg** [comment]
> 🔴 Unable to merge your PR into the base branch. Please rebase or merge it with the base branch.

Говорит нужно rebase сделать

**Review by dorooleg — APPROVED** [review]

**Review by ijon — APPROVED** [review]
schemeshard ok

**Review by gridnevvvit — APPROVED** [review]
